ResourceStatusSystem/Client/SiteStatus.py

```diff
@@ -0,0 +1,247 @@
+
+__RCSID__ = '$Id: $'
+
+class SiteStatus( object ):
+  """
+  RSS helper to interact with the 'Site' family on the DB. It provides the most
+  demanded functions and a cache to avoid hitting the server too often.
+
+  It provides four methods to interact with the site statuses:
+  * getSiteStatuses
+  * getSiteStatus
+  * isUsableSite
+  * getUsableSites
+  """
+
+  __metaclass__ = DIRACSingleton
+
+  def __init__( self ):
+    """
+    Constructor, initializes the rssClient.
+    """
+
+    self.log = gLogger.getSubLogger( self.__class__.__name__ )
+    self.rssConfig = RssConfiguration()
+    self.__opHelper = Operations()
+    self.rssClient = None
+    self.infoGetter = InfoGetter()
+    self.rsClient = ResourceStatusClient()
+
+    # TODO: The RSSCache isn't working right now for sites
+    # We can set CacheLifetime and CacheHistory from CS, so that we can tune them.
+    # cacheLifeTime = int( self.rssConfig.getConfigCache() )
+
+    # RSSCache initialization
+    # self.siteCache  = RSSCache( 'Site', cacheLifeTime, self.__updateSiteCache )
+
+  def getSiteStatuses( self, siteNamesList ):
+    """
+    Method that queries the database for status of the sites in a given list.
+    If the input is None, it is interpreted as * ( all ).
+
+    If match is positive, the output looks like:
+    {
+     'test1.test1.org': 'Active',
+     'test2.test2.org': 'Banned',
+    }
+
+    examples
+      >>> siteStatus.getSiteStatuses( [ 'test1.test1.uk', 'test2.test2.net', 'test3.test3.org' ] )
+          S_OK( { 'test1.test1.org': 'Active', 'test2.test2.net': 'Banned', 'test3.test3.org': 'Active' }  )
+      >>> siteStatus.getSiteStatuses( 'NotExists')
+          S_ERROR( ... ))
+      >>> siteStatus.getSiteStatuses( None )
+          S_OK( { 'test1.test1.org': 'Active',
+                  'test2.test2.net': 'Banned', },
+                  ...
+                }
+              )
+
+    :Parameters:
+      **siteNamesList** - `list`
+        name(s) of the sites to be matched
+
+    :return: S_OK() || S_ERROR()
+    """
+
+    if siteNamesList is None:
+     siteStatusDict = self.rsClient.selectStatusElement( 'Site', 'Status', meta = { 'columns' : [ 'Name', 'Status' ] } )
+
+     if not siteStatusDict['OK']:
+       return S_ERROR(DErrno.ERESGEN, 'selectStatusElement failed')
+     else:
+       siteStatusDict = siteStatusDict['Value']
+
+     return S_OK( dict(siteStatusDict) )
+
+    siteStatusDict = {}
+
+    for siteName in siteNamesList:
+      result = self.rsClient.selectStatusElement( 'Site', 'Status', name = siteName, meta = { 'columns' : [ 'Status' ] } )
+
+      if not result['OK']:
+        return S_ERROR(DErrno.ERESGEN, 'selectStatusElement failed')
+      elif result['Value'] == ():
+        #if one of the listed elements does not exist continue
+        continue
+      else:
+        siteStatusDict[siteName] = result['Value'][0][0]
+
+    return S_OK( siteStatusDict )
+
+
+  def getSiteStatus( self, siteName ):
```

Comment:
Is this function useful? There's already `getSiteStatuses` and this IMHO is just a repetition